sdks/python/apache_beam/runners/direct/evaluation_context.py

```diff
@@ -99,6 +113,19 @@ def finalize_value_and_get_tasks(self, side_input):
       view.has_result = True
       return result
 
+  def update_watermarks_for_transform(self, ptransform, watermark):
+    # Collect tasks that get unblocked as the workflow progresses
```

Comment:
<!--new_thread; commit:2053e91ce5cf41b329a4bf8bdb48a06b79b3626b; resolved:0-->
nit: period at end of sentence.

Reply:
Done